### ISSUE TITLE
how to determine the label of short audio

### ISSUE THREAD

**Issue by brightLLer** [spaghettiSystems/emotion_whisper#1]
It seems that no matter how long the audio is, it will be padded to 30 secs and the model will output a label sequence. If I want to infer on a short audio and just output a single emotion label, what can i do?

**Comment by spaghettiSystems** [comment]
After you downsample your audio to 16KHz get its length (# of samples) and downsample by the same factor as Whisper as follows:
```
effective_length = clip.shape[-1]//160//2
```
And once you perform inference take the mean of the logits 0:effective_length similarly to this
```
torch.mean(extracted_features[:, :length, :], dim=1)
```

This is mentioned in the paper but it's my bad not linking it here ig

**Comment by brightLLer** [comment]
> After you downsample your audio to 16KHz get its length (# of samples) and downsample by the same factor as Whisper as follows:
> 
> ```
> effective_length = clip.shape[-1]//160//2
> ```
> 
> And once you perform inference take the mean of the logits 0:effective_length similarly to this
> 
> ```
> torch.mean(extracted_features[:, :length, :], dim=1)
> ```
> 
> This is mentioned in the paper but it's my bad not linking it here ig

Oh that's awesome! i got it and thank you for your reply
